test/generators/component_generator_test.rb

```diff
@@ -8,7 +8,7 @@
 
 class ComponentGeneratorTest < Rails::Generators::TestCase
   tests Rails::Generators::ComponentGenerator
-  destination File.expand_path("../tmp", __dir__)
+  destination File.expand_path("../../tmp", __dir__)
```

Comment:
Could it make sense to use the OS's tempdir? https://ruby-doc.org/stdlib-2.5.3/libdoc/tmpdir/rdoc/Dir.html#method-c-tmpdir